code/code/misc/discord.h

```diff
@@ -1,19 +1,29 @@
 #pragma once
 
 #include "sstring.h"
+#include <queue>
+#include <thread>
+#include <mutex>
+#include <utility>
 
 class Discord {
-    private: 
-        Discord();
-        static bool sendMessageAsync(sstring channel, sstring msg);
+  private:
+    Discord();
+    static bool sendMessageAsync(sstring channel, sstring msg);
+    static void messenger();
+    static std::thread messenger_thread;
+    static std::mutex queue_mutex;
+    static bool stop_thread;
+    static std::queue<std::pair<sstring, sstring>> message_queue;
```

Comment:
You might (or might not, it's mostly a matter of taste) want to look into the PIMPL way of hiding these implementation details. Check database.h and database.cc for an example.

Reply:
I think I am going to pass on doing it this way.  It seems interesting (and a useful technique), but I don't know if I want to get into those weeds just now.